### DISCUSSION TITLE
[discussion] sb extract

### DISCUSSION THREAD

**Discussion by dvh91** [storybookjs/storybook#13989] [Help]
As for now, the logic behind [extract](https://github.com/storybookjs/storybook/blob/master/lib/cli/src/extract.ts) function in storybook CLI is basically getting the storybook build running in a web server using puppeteer in order to get the stories from the internal client storybook state store.

There're use-cases of using this CLI command in CI environment and it's a pretty much an overkill to get a puppeteer and run the whole build only to get the stories metadata.

The use case I'm using `stories.json` is for newly introduced composed storybook.

Raising this to discussion to see what the community thinks.  I might be available to help out implementing something.

**Comment by shilman** [comment]
AFAIK this is by design. The alternative, JSDOM, has too many corner cases, so we chose Puppeteer to sidestep all of these by running in a real browser.

Hosted services like [Chromatic](https://www.chromatic.com/) also support `stories.json` generation automatically, so you don't have to worry about it.

cc @ndelangen

**Comment by ndelangen** [comment]
@dvh91 we are moving towards having storybook config be declarative & stories be declarative, parameters & decorators & args be declarative.

These are building blocks for being able to extract the stories.json out without using a browser.

We can't do it -right now- but over time we'll hopefully be able to deprecate the non-declarative methods, and extract all data via static analysis.

**Comment by dvh91** [comment]
@ndelangen Thanks for the answer!
Do you know what part of the stories.json is required for the sb composition ([chp level 1](https://storybook.js.org/docs/react/workflows/package-composition#chp-level-1)) ?

trying to figure out if I can, for this use case, use some other solution for the `stories.json` creation.

**Comment by ndelangen** [comment]
Not required, but recommended for performance reasons.

**Comment by dvh91** [comment]
Oh cool! Didn't realize it from the docs.

**Comment by stale** [comment]
Hi everyone! Seems like there hasn't been much going on in this issue lately. If there are still questions, comments, or bugs, please feel free to continue the discussion. Unfortunately, we don't have time to get to every issue. We are always open to contributions so please send us a pull request if you would like to help. Inactive issues will be closed after 30 days. Thanks!

**Comment by pardoman** [comment]
I also echo the sentiment that relying on puppeteer to generate a JSON file is overkill.  I also understand that the developer team is working towards a solution that doesn't include puppeteer, and that's great. 

Could someone link the GitHub Issue # so that we can tracking its progress? Something like "remove puppeteer from sb extract". Thanks.

**Comment by stale** [comment]
Hi everyone! Seems like there hasn't been much going on in this issue lately. If there are still questions, comments, or bugs, please feel free to continue the discussion. Unfortunately, we don't have time to get to every issue. We are always open to contributions so please send us a pull request if you would like to help. Inactive issues will be closed after 30 days. Thanks!

**Comment by pardoman** [comment]
Bump due to stale bot's participation.
